apps/web/Dockerfile

```
@@ -0,0 +1,65 @@
+# Stage 1: Dependency Installation
+# This stage installs only production dependencies for a smaller layer if needed later,
+# or can be merged into the builder stage if devDependencies are small or managed well.
+# For Next.js, devDependencies are often needed for the build, so we'll do a full install in builder.
+FROM node:20-bookworm-slim AS dependencies
+
+WORKDIR /app
+
+# Install pnpm
+RUN npm install -g pnpm
```

Hadolint (2.12.0)

[warning] Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`

(DL3016)